skfp/descriptors/topological.py

```diff
@@ -218,9 +389,57 @@ def zagreb_index(mol: Mol) -> int:
     Examples
     --------
     >>> from rdkit.Chem import MolFromSmiles
-    >>> from skfp.descriptors.topological import zagreb_index
+    >>> from skfp.descriptors.topological import zagreb_index_m1
     >>> mol = MolFromSmiles("C1=CC=CC=C1")  # Benzene
-    >>> zagreb_index(mol)
+    >>> zagreb_index_m1(mol)
     24
     """
     return int(sum(atom.GetDegree() ** 2 for atom in mol.GetAtoms()))
+
+
+def zagreb_index_m2(mol: Mol) -> int:
+    r"""
+    Second Zagreb Index.
+
+    Calculates the second Zagreb index [1]_, defined as the sum of the product of
+    degrees of all pairs of adjacent atoms in the molecule. It provides a measure
+    related to molecular branching and the connectivity of bonds. The formula for
+    calculating the second Zagreb index is given as:
+
+    .. math::
+
+        M_2 = \sum_{(u,v) \in E} d_u \cdot d_v
+
+    where:
+
+    - :math:`E` is the set of bonds (edges) in the molecular graph
+    - :math:`d_u` and :math:`d_v` are the degrees of the atoms :math:`u` and :math:`v`
+      connected by each bond
```

Comment:
`connected by a bond` would be clearer IMO, since "each bond" is already covered by the fact that you have summation int he formula